apps/emqx_durable_storage/src/emqx_ds_storage_bitfield_lts.erl

```diff
@@ -231,14 +231,26 @@ drop(_Shard, DBHandle, GenId, CFRefs, #s{}) ->
 ) ->
     emqx_ds:store_batch_result().
 store_batch(_ShardId, S = #s{db = DB, data = Data}, Messages, _Options) ->
+    {ok, Batch} = rocksdb:batch(),
     lists:foreach(
         fun(Msg) ->
             {Key, _} = make_key(S, Msg),
             Val = serialize(Msg),
-            rocksdb:put(DB, Data, Key, Val, [])
+            rocksdb:batch_put(Batch, Data, Key, Val)
         end,
         Messages
-    ).
+    ),
+    Result = rocksdb:write_batch(DB, Batch, []),
+    rocksdb:release_batch(Batch),
```

Comment:
Should we place this in an `after` clause to ensure it's released?

Reply:
At first glance it seems necessary: calls are not expected to fail with an exception, if they do it's a bug, and the process should probably crash anyway. Which means unreleased batches should be [garbage collected](https://www.erlang.org/doc/man/erl_nif#functionality) since they are NIF resources under the hood.

Reply:
If I understood correctly, it sounds like `enif_release_resource` still needs to be called at some point?

> A resource object is not deallocated until the last handle term is garbage collected by the VM and the resource is released with[ enif_release_resource](https://www.erlang.org/doc/man/erl_nif#enif_release_resource) (not necessarily in that order).

Reply:
It's called [as part of `batch/0`](https://github.com/emqx/erlang-rocksdb/blob/27226d982f5f74965190672bbca5e348a9d9ea94/c_src/batch.cc#L85) actually. The batch itself _supposedly_ should be collected with resource destructor (`batch_resource_cleanup` there) when all refs are dead.

Otherwise it would be impossible to make sure that killed processes won't leak NIF resources.